rclpy/CMakeLists.txt

```diff
@@ -26,19 +26,16 @@ if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
   set(PYTHON_EXECUTABLE "${PYTHON_EXECUTABLE_DEBUG}")
 endif()
 
-# enables using the Python package from the source space
-# This creates a folder in ${CMAKE_CURRENT_BINARY_DIR}/rclpy which has all the
-# as well as the compiled modules from the build space
-# necessary Python code and C Python extensions for running tests.
-configure_file("__init__.py.in" "rclpy/__init__.py" @ONLY)
+# enables using the Python extensions from the build space for testing
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/test_rclpy/__init__.py" "")
 
 ament_python_install_package(${PROJECT_NAME})
 
 function(set_properties _targetname _build_type)
   set_target_properties(${_targetname} PROPERTIES
     PREFIX ""
-    LIBRARY_OUTPUT_DIRECTORY${_build_type} "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
-    RUNTIME_OUTPUT_DIRECTORY${_build_type} "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}"
+    LIBRARY_OUTPUT_DIRECTORY${_build_type} "${CMAKE_CURRENT_BINARY_DIR}/test_${PROJECT_NAME}"
+    RUNTIME_OUTPUT_DIRECTORY${_build_type} "${CMAKE_CURRENT_BINARY_DIR}/test_${PROJECT_NAME}"
```

Comment:
It looks a bit weird to me that if I build without build_testing all my rclpy libraries go to this `test_rclpy` folder. Then they'll be install at the right place at the end of the day so I guess it's fine

Reply:
The directory can have any name except `rclpy`. I used the `test_` prefix since it is being used during testing. But any other name (which isn't likely to collide with other packages) would be fine.